extensions/amp-story/1.0/amp-story-store-service.js

```diff
@@ -148,6 +151,16 @@ export const Action = {
 };
 
 
+/**
+ * Functions to compare a data structure from the previous to the new state and
+ * detect a mutation, when a simple equality test would not work.
+ * @private @const {!Object<string, !function(*, *):boolean>}
+ */
+const stateComparisonFunctions = {
+  [StateProperty.ACTIONS_WHITELIST]: (old, curr) => old.length !== curr.length,
+};
```

Comment:
This works. If state values are all JSON, another option is to use `json.js#deepEquals()` with depth == 2.

Reply:
This method is pretty cool I had no idea we had this

Reply:
It's also kind of slow for very large JSON (depth >= 10).